llvm_passes/HipIGBADetector.cpp

```diff
@@ -0,0 +1,102 @@
+//===- HipIGBADetector.cpp --------------------------------------===//
+//
+// Part of the chipStar Project, under the Apache License v2.0 with LLVM
+// Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This pass analyzes module to find out if it has potential indirect global
+// buffer accesses (IGBA). The outcome of the analysis is stored in a magic
+// variable for the chipStar runtime:
+//
+//   uint8_t __chip_module_may_have_IGBAs = <result>;
```

Comment:
Perhaps use __chipStar_ as the prefix. 'chip' is misleading.

Reply:
We are using `__chip_` prefix in a lot of places so I think I'll keep it for style consistency.